internal/runners/clean/config.go

```diff
@@ -53,12 +54,13 @@ func (c *Config) Run(params *ConfigParams) error {
 	}
 
 	if !params.Force {
-		ok, err := c.confirm.Confirm(locale.T("confirm"), locale.T("clean_config_confirm"), new(bool))
+		defaultChoice := params.NonInteractive
+		ok, err := c.confirm.Confirm(locale.T("confirm"), locale.T("clean_config_confirm"), &defaultChoice)
```

Comment:
This is already handled: https://github.com/ActiveState/cli/blob/52511befe457f17b63048fdc07b78be9d50c861f/internal/prompt/prompt.go#L149

You can just pass `p.BoolP(true)`. The default is used when ran under non-interactive mode.

In fact I think you can reduce this PR to pretty much this one change. You shouldn't need to pass NonInteractive to the runner.

Reply:
But we want the default to be `false` if running interactively. If I pass `p.BoolP(true)`, then running `state clean config` prompts me with `Y/n` (Y) as the default. What am I missing?

Reply:
> But we want the default to be false if running interactively

I think you're asserting that based on what the code currently does? I'm not seeing anything on the Jira ticket about this.

Imo the default should just be true, regardless of what mode you're running under. This is how we've handled all non-interactive prompts so far. The reason why there are some prompts with the wrong default is because the prompts preceded the non-interactive mechanic, so not all of them are well curated towards the non-interactive mechanic.